streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -364,6 +365,16 @@ private void commitOffsets(final boolean startNewTransaction) {
         }
     }
 
+    Map<TopicPartition, Long> purgableOffsets() {
+        final Map<TopicPartition, Long> purgableConsumedOffsets = new HashMap<>();
+        for (Map.Entry<TopicPartition, Long> entry : consumedOffsets.entrySet()) {
+            if (topology.isTopicInternalTransient(entry.getKey().topic()))
+                purgableConsumedOffsets.put(entry.getKey(), entry.getValue() + 1);
```

Comment:
is that right to purge the consumed offset + 1?

Reply:
> is that right to purge the consumed offset + 1?

The delete records call is to delete records `before` the given offset, similar to commit offset, so this is correct.